core/src/test/scala/org/broadinstitute/dsde/rawls/deltalayer/GcsDeltaLayerWriterSpec.scala

```diff
@@ -0,0 +1,118 @@
+package org.broadinstitute.dsde.rawls.deltalayer
+
+import org.broadinstitute.dsde.rawls.TestExecutionContext
+import org.broadinstitute.dsde.workbench.model.google.GcsBucketName
+import org.scalatest.flatspec.AnyFlatSpec
+import TestExecutionContext.testExecutionContext
+import akka.actor.ActorSystem
+import com.google.cloud.storage.Storage.BlobWriteOption
+import com.google.cloud.storage.contrib.nio.testing.LocalStorageHelper
+import com.google.cloud.storage.{BlobInfo, Storage, StorageException}
+import org.broadinstitute.dsde.rawls.model.AttributeUpdateOperations.{AddUpdateAttribute, AttributeUpdateOperation, EntityUpdateDefinition}
+import org.broadinstitute.dsde.rawls.model.{AttributeName, AttributeString, DeltaInsert, Destination, RawlsUserSubjectId}
+import org.broadinstitute.dsde.workbench.google2.GcsBlobName
+import org.joda.time.DateTime
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.RecoverMethods.recoverToExceptionIf
+import org.scalatest.concurrent.Eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatestplus.mockito.MockitoSugar.mock
+
+import java.util.UUID
+import scala.util.Try
+
+class GcsDeltaLayerWriterSpec extends AnyFlatSpec with GcsStorageTestSupport with Eventually with Matchers {
+
+  implicit val actorSystem: ActorSystem = ActorSystem("GcsDeltaLayerWriterSpec")
+
+  behavior of "GcsDeltaLayerWriter"
+
+  it should "write the specified file to the bucket" in {
+    val bucket = GcsBucketName("successful-write")
+
+    // explicitly create the storage so we can query it directly
+    val storage = LocalStorageHelper.getOptions.getService
+    val deltaLayerWriter = getGcsWriter(bucket, Some(storage))
```

Comment:
`LocalStorageHelper` is provided by Google to assist with testing; it mocks the real cloud-based storage. See https://javadoc.io/doc/com.google.cloud/google-cloud-nio/latest/com/google/cloud/storage/contrib/nio/testing/LocalStorageHelper.html